internal/juju/machines.go

```diff
@@ -89,8 +92,51 @@ func (c machinesClient) CreateMachine(ctx context.Context, input *CreateMachineI
 	}
 	defer func() { _ = conn.Close() }()
 
-	machineAPIClient := apimachinemanager.NewClient(conn)
+	response, err := c.createMachine(ctx, conn, input)
+	if err != nil {
+		return nil, err
+	}
+
+	// Wait for machine to go into "running state".
+	retryErr := retry.Call(retry.CallArgs{
+		Func: func() error {
+			modelStatus, err := c.ModelStatus(input.ModelName, conn)
+			if err != nil {
+				return errors.Annotatef(err, "failed to get model status.")
+			}
+			machineStatus, ok := modelStatus.Machines[response.ID]
```

Comment:
question: Depending on the version of juju, it may be possible to run this code before status has the new machine in it. The juju model cache for status wasn't fully removed until late juju 3 versions? We'd want to double check that.

Reply:
i see.. i'll return a retry error